ui/app/components/app/metamask-translation/metamask-translation.js

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import MetaMaskTemplateRenderer, {
+  SectionShape,
+} from '../metamask-template-renderer/metamask-template-renderer';
+
+/**
+ * MetaMaskTranslation is a simple helper Component for adding full translation
+ * support to the template system. We do pass the translation function to the
+ * template getValues function, but passing it react components as variables
+ * would require react to be in scope, and breaks the object pattern paradigm.
```

Comment:
I'm not entirely sure what you mean by React being in-scope, or why it might be a problem. Are you referring just to the import? We can do away with that by updating to React v17, so I'm not too worried about that.

Breaking the object pattern paradigm though, that part I understand.

Reply:
i don't really want to have JSX at all in the template files, because ultimately they may not even exist/be generated in the same app that has our React in it. Having React in scope is not the primary concern here and I didn't' realize v17 removed the necessity for importing it!

Reply:
>i don't really want to have JSX at all in the template files, because ultimately they may not even exist/be generated in the same app that has our React in it.

Ah, right, good point! Makes sense.